src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeMemory.Unix.cs

```diff
@@ -0,0 +1,149 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Numerics;
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>This class contains methods that are mainly used to manage native memory.</summary>
+    public static unsafe class NativeMemory
+    {
+        /// <summary>Allocates an aligned block of memory of the specified size and alignment, in bytes.</summary>
+        /// <param name="byteCount">The size, in bytes, of the block to allocate. This should be a multiple of <paramref name="alignment" /> for portability.</param>
+        /// <param name="alignment">The alignment, in bytes, of the block to allocate. This must be a power of <c>2</c> greater than or equal to <c>sizeof(void*)</c> for portability.</param>
+        /// <returns>A pointer to the allocated aligned block of memory.</returns>
+        /// <exception cref="OutOfMemoryException">Allocating <paramref name="byteCount" /> of memory failed.</exception>
+        /// <remarks>
+        ///     <para>This method allows <paramref name="byteCount" /> to be <c>0</c> and will return a valid pointer that should not be dereferenced and that should be passed to free to avoid memory leaks.</para>
+        ///     <para>This method is a thin wrapper over the C <c>aligned_alloc</c> API or a platform dependent aligned allocation API such as <c>_aligned_malloc</c> on Win32.</para>
+        ///     <para>This method is not compatible with <see cref="Free" /> or <see cref="Realloc" />, instead <see cref="AlignedFree" /> should be called.</para>
+        /// </remarks>
+        [CLSCompliant(false)]
+        public static void* AlignedAlloc(nuint byteCount, nuint alignment)
```

Comment:
I'm wondering if we should have these be `AggressiveInlining`. These functions, particularly with the `throw`, are likely going to be "too expensive" to inline otherwise

But the actual failure cases are likely going to be rare (particularly when constants are input) and these functions are already incurring the P/Invoke transition overhead (for something which is generally heavily optimized to be "fast" for the common case) and so having all these indirections seems undesirable to me.

Reply:
> AggressiveInlining

This would need to be backed by data.